src/lib/rosetta_lib/transaction.ml

```diff
@@ -446,3 +446,70 @@ module Signed = struct
           (Errors.create ~context:"Signed transaction un-rendering"
              `Unsupported_operation_for_construction)
 end
+
+let to_mina_signed transaction_json =
+  Or_error.try_with_join (fun () ->
+      let open Or_error.Let_syntax in
+      let%bind rosetta_transaction_rendered =
+        Signed.Rendered.of_yojson transaction_json
+        |> Result.map_error ~f:Error.of_string
+      in
+      let%bind rosetta_transaction =
+        Signed.of_rendered rosetta_transaction_rendered
+        |> Result.map_error ~f:(fun err -> Error.of_string (Errors.show err))
+      in
+      let valid_until, memo =
+        (* This is a hack..
+           TODO: Handle these properly in rosetta.
+        *)
+        match rosetta_transaction.command.kind with
+        | `Payment ->
+            ( Option.bind rosetta_transaction_rendered.payment
+                ~f:(fun {valid_until; _} -> valid_until)
+            , Option.bind rosetta_transaction_rendered.payment
+                ~f:(fun {memo; _} -> memo) )
+        | `Delegation ->
+            ( Option.bind rosetta_transaction_rendered.stake_delegation
+                ~f:(fun {valid_until; _} -> valid_until)
+            , Option.bind rosetta_transaction_rendered.stake_delegation
+                ~f:(fun {memo; _} -> memo) )
+        | `Create_token ->
+            ( Option.bind rosetta_transaction_rendered.create_token
+                ~f:(fun {valid_until; _} -> valid_until)
+            , Option.bind rosetta_transaction_rendered.create_token
+                ~f:(fun {memo; _} -> memo) )
+        | `Create_token_account ->
+            ( Option.bind rosetta_transaction_rendered.create_token_account
+                ~f:(fun {valid_until; _} -> valid_until)
+            , Option.bind rosetta_transaction_rendered.create_token_account
+                ~f:(fun {memo; _} -> memo) )
+        | `Mint_tokens ->
+            ( Option.bind rosetta_transaction_rendered.mint_tokens
+                ~f:(fun {valid_until; _} -> valid_until)
+            , Option.bind rosetta_transaction_rendered.mint_tokens
+                ~f:(fun {memo; _} -> memo) )
+      in
+      let pk (`Pk x) =
+        Signature_lib.Public_key.Compressed.of_base58_check_exn x
```

Comment:
Should this be an error, if the decoding fails?

Reply:
It should be already, it gets caught as one in the `Or_error.try_with_join` above